snippets/paths.py

```diff
@@ -1,47 +1,101 @@
-"""Example of how to find the best path to trade with"""
+"""Example of ripple_path_find request using trustlines and rippling"""
+
 from xrpl.clients import JsonRpcClient
-from xrpl.models import XRP, IssuedCurrencyAmount, Payment, RipplePathFind
-from xrpl.transaction import autofill_and_sign
+from xrpl.models import AccountSet, TrustSet
+from xrpl.models.amounts import IssuedCurrencyAmount
+from xrpl.models.requests import RipplePathFind
+from xrpl.models.transactions.account_set import AccountSetAsfFlag
+from xrpl.transaction import submit_and_wait
 from xrpl.wallet import generate_faucet_wallet
 
-# References
-# - https://xrpl.org/paths.html#paths
-# - https://xrpl.org/ripple_path_find.html#ripple_path_find
+# Note: This test is inspired from a unit test titled `indirect_paths_path_find` in the
+# rippled C++ codebase (Path_test.cpp)
+# https://github.com/XRPLF/rippled/blob/d9bd75e68326861fb38fd5b27d47da1054a7fc3b/src/test/app/Path_test.cpp#L683
 
 # Create a client to connect to the test network
 client = JsonRpcClient("https://s.altnet.rippletest.net:51234")
 
 # Creating wallet to send money from
-wallet = generate_faucet_wallet(client, debug=True)
-
-# Create account and amount variables for later transaction
-destination_account = "rKT4JX4cCof6LcDYRz8o3rGRu7qxzZ2Zwj"
-destination_amount = IssuedCurrencyAmount(
-    value="0.001",
-    currency="USD",
-    issuer="rVnYNK9yuxBz4uP8zC8LEFokM2nqH3poc",
+# these wallets will have 100 testnet XRP
+alice = generate_faucet_wallet(client, debug=True)
+bob = generate_faucet_wallet(client, debug=True)
+carol = generate_faucet_wallet(client, debug=True)
+
+# send AccountSet transaction with asfDefaultRipple turned on
+# this enables rippling on all trustlines through these accounts.
+submit_and_wait(
+    AccountSet(account=alice.address, set_flag=AccountSetAsfFlag.ASF_DEFAULT_RIPPLE),
+    client,
+    alice,
 )
 
-# Create a RipplePathFind request and have the client call it
-path_request = RipplePathFind(
-    source_account=wallet.address,
-    source_currencies=[XRP()],
-    destination_account=destination_account,
-    destination_amount=destination_amount,
+submit_and_wait(
+    AccountSet(account=bob.address, set_flag=AccountSetAsfFlag.ASF_DEFAULT_RIPPLE),
+    client,
+    bob,
 )
-path_response = client.request(path_request)
-print(path_response)
 
-# Extract out paths from the RipplePathFind response
-paths = path_response.result["alternatives"][0]["paths_computed"]
-print(paths)
+submit_and_wait(
```

Comment:
Before these `submit_and_wait`, can you print out a statement such as Submitting AccountSet transaction for ...` so that user can expect to wait, and also print result or some kind of message when done?